test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
```

Comment:
To be clear: All `stdpath()` does is string manipulation.  Testing that `~/foo` and `$BAR` aren't expanded would be in the tests for `stdpaths_get_xdg_var()` (see below).

If someone points me at the right direction, I can see about creating a test or 3 for `stdpaths_get_xdg_var()`.